contracts/strategies/alkemi/AlkemiStrategy.sol

```diff
@@ -0,0 +1,304 @@
+import "../../base/upgradability/BaseUpgradeableStrategyUL.sol";
+import "../../base/interface/uniswap/IUniswapV2Pair.sol";
+import "./interface/IAlkemiEarnPublic.sol";
+import "./interface/IRewardControl.sol";
+import "../../base/interface/weth/Weth9.sol";
+
+contract AlkemiStrategy is IStrategy, BaseUpgradeableStrategyUL {
+
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  address public constant wethAddr = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+  address public constant multiSigAddr = address(0xF49440C1F012d041802b25A73e5B0B9166a75c02);
+
+  // additional storage slots (on top of BaseUpgradeableStrategy ones) are defined here
+  bytes32 internal constant _ALKEMI_TOKEN_SLOT = 0x64ebd51c4e794221918d0decf03cb433a69ffbab192707e00b41ef0ce07f6217;
+  bytes32 internal constant _SUPPLY_CONTRACT_SLOT = 0x52fbc10413f749a4bd118a304e2a9446461114a4bb64b3a69cdb9cca463f1e81;
+  bytes32 internal constant _HODL_RATIO_SLOT = 0xb487e573671f10704ed229d25cf38dda6d287a35872859d096c0395110a0adb1;
+  bytes32 internal constant _HODL_VAULT_SLOT = 0xc26d330f887c749cb38ae7c37873ff08ac4bba7aec9113c82d48a0cf6cc145f2;
+
+  uint256 public constant hodlRatioBase = 10000;
+
+  constructor() public BaseUpgradeableStrategyUL() {
+    assert(_ALKEMI_TOKEN_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.alkemiToken")) - 1));
+    assert(_SUPPLY_CONTRACT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.supplyContract")) - 1));
+    assert(_HODL_RATIO_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlRatio")) - 1));
+    assert(_HODL_VAULT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlVault")) - 1));
+  }
+
+  function initializeBaseStrategy(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    address _rewardToken,
+    address _alkemiToken,
+    address _supplyContract,
+    uint256 _hodlRatio
+  ) public initializer {
+    uint256 psNum = 300;
+    if (_hodlRatio >= 3000) {
+      psNum = 0;
+    } else if (_hodlRatio > 0){
+      psNum = psNum.sub(_hodlRatio.div(10)).mul(hodlRatioBase).div(hodlRatioBase.sub(_hodlRatio));
+    }
+    BaseUpgradeableStrategyUL.initialize(
+      _storage,
+      _underlying,
+      _vault,
+      _rewardPool,
+      _rewardToken,
+      psNum,  // profit sharing numerator
+      1000, // profit sharing denominator
+      true, // sell
+      1e15, // sell floor
+      12 hours, // implementation change delay
+      address(0x7882172921E99d590E097cD600554339fBDBc480) //UL Registry
+    );
+    _setAlkemiToken(_alkemiToken);
+    _setSupplyContract(_supplyContract);
+    setHodlRatio(_hodlRatio);
+    setHodlVault(multiSigAddr);
+  }
+
+  function setHodlRatio(uint256 _value) public onlyGovernance {
+    uint256 psNum = 300;
+    if (_value >= 3000) {
+      psNum = 0;
+    } else if (_value > 0){
+      psNum = psNum.sub(_value.div(10)).mul(hodlRatioBase).div(hodlRatioBase.sub(_value));
```

Comment:
1. `psNum` -> `profitSharingNumerator`
2. Let's break down this line for readability.

Reply:
Updated, also added check on `setHodlRatio` to not allow higher value than base